src/services/ldap/index.js

```diff
@@ -1,250 +1,272 @@
-					if (err) {
-						reject(err);
-					}
-					res.on('error', reject);
-					res.on('searchEntry', (entry) => {
-						const result = entry.object;
-						rawAttributes.forEach((element) => {
-							result[element] = entry.raw[element].toString('base64');
-						});
-						objects.push(result);
-					});
-					res.on('end', (result) => {
-						if (result.status === 0) {
-							resolve(objects);
-						}
-						reject(new Error('LDAP result code != 0'));
-					});
-				});
-			}));
-		}
-
-		/**
-         * Returns first LDAP object matching the given search string and options
-         * @see searchCollection
-         * @param {LdapConfig} config the ldapConfig
-         * @param {String} searchString the search string
-         * @param {Object} options search options
-         * @return {Promise[Object]} resolves with object matching the query,
-         * rejects with error otherwise
-         */
-		searchObject(config, searchString, options) {
-			return this.searchCollection(config, searchString, options)
-				.then((objects) => {
-					if (objects.length > 0) {
-						return Promise.resolve(objects[0]);
-					}
-					return Promise.reject(new Error(`Object "${searchString}" not found`));
-				});
-		}
-
-		/**
-         * Returns all schools on the LDAP server
-         * @param {LdapConfig} config the ldapConfig
-         * @return {Promise[Array[Object]]} resolves with all school objects or
-         * rejects with error
-         */
-		getSchools(config) {
-			return getLDAPStrategy(app, config).getSchools();
-		}
-
-		/**
-         * Returns all users at a school on the LDAP server
-         * @param {LdapConfig} config the ldapConfig
-         * @param {School} school the school object
-         * @return {Promise[Object]} resolves with all user objects or rejects
-         * with error
-         */
-		getUsers(config, school) {
-			return getLDAPStrategy(app, config).getUsers(school);
-		}
-
-		/**
-         * Returns all classes at a school on the LDAP server
-         * @param {LdapConfig} config the ldapConfig
-         * @param {School} school the school object
-         * @return {Promise[Object]} resolves with all class objects or rejects
-         * with error
-         */
-		getClasses(config, school) {
-			return getLDAPStrategy(app, config).getClasses(school);
-		}
-
-		/**
-         * Returns all experts on the LDAP server
-         * @param {LdapConfig} config the ldapConfig
-         * @return {Promise[Object]} resolves with all expert objects or rejects
-         * with error
-         */
-		getExperts(config) {
-			const { searchString, options } = getLDAPStrategy(app, config).getExpertsQuery();
-			return this.searchCollection(config, searchString, options);
-		}
-	}
-
-	app.use('/ldap', new LdapService());
-	const systemService = app.service('/ldap');
-	systemService.hooks(hooks);
-};
+/* eslint-disable no-underscore-dangle */
+const ldap = require('ldapjs');
+const errors = require('@feathersjs/errors');

```

Comment:
Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5713943)